frontend/openchat-agent/src/services/common/chatMappers.ts

```diff
@@ -1256,8 +1266,57 @@ export function communitySummary(candid: ApiCommunityCanisterCommunitySummary):
         membership: {
             joined: optional(candid.membership, (m) => m.joined) ?? BigInt(0),
             role: optional(candid.membership, (m) => memberRole(m.role)) ?? "none",
+            archived: false, // TODO not sure what to do about this
+            pinned: [], // TODO also not sure about this - comes from the user canister
+        },
+        channels: candid.channels.map((c) => communityChannelSummary(c, communityId)),
+    };
+}
+
+export function communityChannelSummary(
+    candid: ApiCommunityCanisterChannelSummary,
+    communityId: string
+): ChannelSummary {
+    const latestMessage = optional(candid.latest_message, (ev) => ({
+        index: ev.index,
+        timestamp: ev.timestamp,
+        event: message(ev.event),
+    }));
+    return {
+        kind: "channel",
+        id: { kind: "channel", communityId, channelId: candid.channel_id.toString() },
+        latestMessage,
+        name: candid.name,
+        description: candid.description,
+        public: candid.is_public,
+        historyVisible: candid.history_visible_to_new_joiners,
+        minVisibleEventIndex: candid.min_visible_event_index,
+        minVisibleMessageIndex: candid.min_visible_message_index,
+        latestEventIndex: candid.latest_event_index,
+        lastUpdated: candid.last_updated,
+        blobReference: optional(candid.avatar_id, (blobId) => ({
+            blobId,
+            canisterId: candid.channel_id.toString(),
+        })),
+        memberCount: candid.member_count,
+        permissions: groupPermissions(candid.permissions),
+        metrics: chatMetrics(candid.metrics),
+        subtype: optional(candid.subtype, apiGroupSubtype),
+        frozen: false, // TODO - doesn't exist
+        dateLastPinned: optional(candid.date_last_pinned, identity),
+        dateReadPinned: undefined,
+        gate: optional(candid.gate, accessGate) ?? { kind: "no_gate" },
+        level: "group",
+        membership: {
+            joined: optional(candid.membership, (m) => m.joined) ?? BigInt(0),
+            notificationsMuted: false,
+            role: optional(candid.membership, (m) => memberRole(m.role)) ?? "none",
+            myMetrics: emptyChatMetrics(),
+            readByMeUpTo: latestMessage?.event.messageIndex,
```

Comment:
This should come from the user canister

Reply:
it _will_ do when we go through the merge process in openChatAgent